bin/inx-chronicle/src/launcher.rs

```diff
@@ -75,15 +80,20 @@ impl HandleEvent<Report<super::stardust_inx::InxWorker>> for Launcher {
         &mut self,
         cx: &mut ActorContext<Self>,
         event: Report<super::stardust_inx::InxWorker>,
-        config: &mut Self::State,
+        LauncherState { config, db }: &mut Self::State,
     ) -> Result<(), Self::Error> {
+        use super::stardust_inx::InxError;
         match event {
             Report::Success(_) => {
                 cx.abort().await;
             }
             Report::Error(report) => match report.error {
                 ActorError::Result(e) => match e {
-                    super::stardust_inx::InxError::MongoDb(e) => match e.kind.as_ref() {
+                    InxError::ConnectionError(_) | InxError::TransportFailed(_) => {
```

Comment:
We should only retry on `InxError::ConnectionError`, if `TransportFailed` we should report an `ERROR` and shutdown Chronicle.